app/javascript/components/users/UserEvents.vue

```diff
@@ -0,0 +1,99 @@
+                    <div class="col-2" @click="alert('h')">
+                        <button class="btn-warning edit-event-button">Edit event</button>
+                    </div>
+                    <div class="col-2">
+                        <p class="event-status" :class="event.status">{{ event.status }}</p>
+                    </div>
+                </div>
+                <div v-bind:id="'event'+event.id" class="display-none">
+                    <p v-for="title in Object.keys(titlesEvent)" v-if="event[titlesEvent[title]]!==null">{{ title +" - " + event[titlesEvent[title]] }}</p>
+                    <div class="row" v-if="event.targets.length!==0">
+                        <div class="col-2">Targets:</div>
+                        <div class="col-10">
+                            <div class="row" v-for="target in event.targets">
+                                <div v-if="axiosFlashNotice" class="alert alert-danger flash_notice">{{ axiosFlashNotice }}</div>
+                                <div class="col-10">
+                                    <label v-bind:for="'target'+target.id">{{ target.description }}</label>
+                                </div>
+                                <div class="col-2">
+                                    <input v-if="target.achieved" v-bind:id="'target'+target.id" type="checkbox" class="checkbox" checked @click="changeTargetAchieved(target.id)">
+                                    <input v-else type="checkbox" v-bind:id="'target'+target.id" class="checkbox" @click="changeTargetAchieved(target.id)">
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+    export default {
+        data: function () {
+            return {
+                events: JSON.parse(this.eventsForProps),
+                topYear: null,
+                axiosFlashNotice: false,
+                titlesEvent: {
+                    "Comments": "comments",
+                    "Summary": "summary",
+                    "Start date": "start_date",
+                    "Finish date": "finish_date",
+                },
+            }
+        },
+        props: {
+            eventsForProps: { type: String },
+            changeTargetAchievedPath: { type: String }
+        },
+        methods: {
+            getFullYear(event) {
+                return new Date(event['finish_date']).getFullYear()
+            },
+            topEqualCurrent(event) {
+                if (this.getFullYear(event) === this.topYear ){
+                    return true
+                }
+                else {
+                    this.topYear = this.getFullYear(event);
+                    return false
+                }
+            },
+            changeActiveEvent (eventID){
+                let hidden_event_info = document.getElementById('event'+eventID);
```

Comment:
Use camelCase for variables: https://github.com/airbnb/javascript#naming--camelCase

Reply:
Add missed spaces